### PR TITLE
Fix ValueError(ndim of gamma and beta) of batch normalization when using Theano

### PR DESCRIPTION
**Pull request by woozzu**
I am using the latest version of theano and keras, and ValueError same as #3667 occurs when using batch normalization. The error is caused by theano's batch normalization methods, which require the exactly same ndim as input. So, in all cases gamma and beta must be reshaped when using theano.

### CODE REVIEW

**Comment by fchollet** [comment]
This sounds like more of a Theano bug, and it was definitely not the case in earlier versions of Theano. Shouldn't Theano be able to automatically do the broadcast?

**Comment by woozzu** [comment]
@fchollet In earlier versions, batch norm implementation of Theano is called instead of that of cuDNN. See [theano_backend.py#419](https://github.com/fchollet/keras/blob/master/keras/backend/theano_backend.py#L419). `theano.sandbox.cuda.dnn.dnn_batch_normalization_test` is in the latest version, and the error occurs in that method. While `dnn_batch_normalization_test` strictly checks the dimensionality of parameters, `T.nnet.bn.batch_normalization` doesn't. (ps. I think `T.nnet.bn.batch_normalization` is more buggy because it doesn't check parameters and users mistake.)

The purpose of if statement in [normalization.py#143](https://github.com/fchollet/keras/blob/master/keras/layers/normalization.py#L143) is to automatically broadcast the last depth dimension. However, we cannot be sure the last dimension is always depth in Theano. So, it's not a common case in Theano compared to Tensorflow (See [tensorflow's batch norm](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/api_docs/python/functions_and_classes/shard8/tf.nn.batch_normalization.md)). In my opinion, it's more general to always broadcast all dimensions before batch normalization.

**Comment by alexmarkley** [comment]
I wanted to +1 this. It fixes Issue #3667 for me.

File: keras/layers/normalization.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def call(self, x, mask=None):
                 self.updates = [K.moving_average_update(self.running_mean, mean, self.momentum),
                                 K.moving_average_update(self.running_std, std, self.momentum)]
 
-                if sorted(reduction_axes) == range(K.ndim(x))[:-1]:
+                if K._BACKEND == 'tensorflow' and sorted(reduction_axes) == range(K.ndim(x))[:-1]:
```
Comment (fchollet) [inline keras/layers/normalization.py]: The proper way to check would be `K.backend() == 'tensorflow'`.

Comment (woozzu) [inline keras/layers/normalization.py]: @fchollet Thanks. I fixed it.

**Comment by fchollet** [comment]
LGTM
